src/Entity/TaskRepository.php

```diff
@@ -73,4 +75,26 @@ public function findBySubscriptionAndDate(Task\RecurrenceRule $subscription, Dat
             ->getQuery()
             ->getResult();
     }
+
+    public function findByBarcode(string $barcode): ?Task
+    {
+        $barcode = BarcodeUtils::parse($barcode);
+
+        if ($barcode->isInternal() && $barcode->getEntityType() === Barcode::TYPE_TASK) {
+            return $this->find($barcode->getEntityId());
+        }
+
+        /* TODO: I'm using raw sql here, but doctrine doesn't support json fields and i don't
+                 want to install doctrine json functions just for this
+            @see https://github.com/ScientaNL/DoctrineJsonFunctions
+        */
+        $conn = $this->getEntityManager()->getConnection();
+        $sql = "SELECT id FROM task WHERE metadata->>'barcode' = :barcode";
```

Comment:
Check if you get the correct function here https://github.com/martin-georgiev/postgresql-for-doctrine/tree/main/src/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions

then add the function like this https://github.com/coopcycle/coopcycle-web/blob/master/app/config/config.yml#L335